relay-server/src/actors/project.rs

```diff
@@ -522,29 +532,45 @@ impl Project {
             state_channel: None,
             rate_limits: RateLimits::new(),
             last_no_cache: Instant::now(),
-            metrics_allowed: true,
         }
     }
 
     /// If we know that a project is disabled, disallow metrics, too.
-    fn update_metrics_allowed(&mut self) {
-        if let Some(state) = self.state() {
-            self.metrics_allowed = state.check_disabled(&self.config).is_ok();
+    fn metrics_allowed(&self) -> bool {
+        if let Some(state) = self.get_valid_state() {
+            state.check_disabled(&self.config).is_ok()
+        } else {
+            // Projects without state go back to the original state of allowing metrics.
+            true
         }
     }
 
     pub fn merge_rate_limits(&mut self, rate_limits: RateLimits) {
         self.rate_limits.merge(rate_limits);
     }
 
-    /// Returns a reference to the project state if available.
-    pub fn state(&self) -> Option<&ProjectState> {
-        self.state.as_deref()
+    /// Returns the current [`ExpiryState`] for this project.
+    /// If the project state's [`Expiry`] is `Expired`, do not return it.
+    pub fn get_expiry_state(&self) -> ExpiryState {
```

Comment:
Getters in Rust usually do not have a `get_` prefix.

```suggestion
    pub fn expiry_state(&self) -> ExpiryState {
```